src/test/java/com/exasol/MetadataReadingIT.java

```diff
@@ -61,27 +60,85 @@ void testDetermineColumnTypes() {
-                        .row("CU", "CHAR(345) UTF8", anything(), anything(), anything())
-                        .row("DA", "DATE", anything(), anything(), anything())
-                        .row("DO", "DOUBLE", anything(), anything(), anything())
-                        .row("DE", "DECIMAL(15,9)", anything(), anything(), anything())
-                        .row("G1", "GEOMETRY(7)", anything(), anything(), anything())
-                        .row("G2", "GEOMETRY", anything(), anything(), anything())
-                        .row("H1", "HASHTYPE(4 BYTE)", anything(), anything(), anything())
-                        .row("H2", "HASHTYPE(20 BYTE)", anything(), anything(), anything())
-                        .row("I1", "INTERVAL YEAR(2) TO MONTH", anything(), anything(), anything())
-                        .row("I2", "INTERVAL YEAR(7) TO MONTH", anything(), anything(), anything())
-                        .row("I3", "INTERVAL DAY(2) TO SECOND(3)", anything(), anything(), anything())
-                        .row("I4", "INTERVAL DAY(4) TO SECOND(2)", anything(), anything(), anything())
-                        .row("T1", "TIMESTAMP", anything(), anything(), anything())
-                        .row("T2", "TIMESTAMP WITH LOCAL TIME ZONE", anything(), anything(), anything())
-                        .row("VA", "VARCHAR(123) ASCII", anything(), anything(), anything())
-                        .row("VU", "VARCHAR(12) UTF8", anything(), anything(), anything()) //
-                        .matches());
+        final User user = createUserWithVirtualSchemaAccess("USER_COLUMN_TYPE", virtualSchema);
+        assertVirtualTableStructure(table, user, expectRows("BO", "BOOLEAN", //
+                "CA", "CHAR(34) ASCII", //
+                "CU", "CHAR(345) UTF8", //
+                "DA", "DATE", //
+                "DO", "DOUBLE", //
+                "DE", "DECIMAL(15,9)", //
+                "G1", "GEOMETRY(7)", //
+                "G2", "GEOMETRY", //
+                "H1", "HASHTYPE(4 BYTE)", //
+                "H2", "HASHTYPE(20 BYTE)", //
+                "I1", "INTERVAL YEAR(2) TO MONTH", //
+                "I2", "INTERVAL YEAR(7) TO MONTH", //
+                "I3", "INTERVAL DAY(2) TO SECOND(3)", //
+                "I4", "INTERVAL DAY(4) TO SECOND(2)", //
+                "T1", "TIMESTAMP", //
+                "T2", "TIMESTAMP WITH LOCAL TIME ZONE", //
+                "VA", "VARCHAR(123) ASCII", //
+                "VU", "VARCHAR(12) UTF8"));
+    }
+
+    private void assertVirtualTableStructure(final Table table, final User user,
+            final Matcher<ResultSet> tableMatcher) {
+        assertRlsQueryWithUser("DESCRIBE " + getVirtualSchemaName(table.getParent().getName()) + "." + table.getName(),
+                user, //
+                tableMatcher);
+    }
+
+    private Matcher<ResultSet> expectRows(final String... strings) {
+        assert ((strings.length % 2) == 0);
```

Comment:
better use a junit assertion with a meaningful error message in case `-ea` is not used.